File: ai_tnhn/ai-api-tnhn/router/station.go
package router

import (
	"ai-api-tnhn/handler"
	"ai-api-tnhn/router/middleware"

	"github.com/gin-gonic/gin"
)

func (h HandlerFuncs) StationRoutes(api *gin.RouterGroup, mid middleware.Middleware, stationHandler *handler.StationHandler, pumpingHandler *handler.PumpingStationHandler, wastewaterHandler *handler.WastewaterTreatmentHandler, sluiceHandler *handler.SluiceGateHandler) {
	station := api.Group("/stations")
	station.Use(mid.MidBasicType())

	// Rain Stations
	rain := station.Group("/rain")
	{
		rain.GET("", stationHandler.ListRain)
		rain.GET("/:id", stationHandler.GetRainByID)
		rain.POST("", stationHandler.CreateRain)
		rain.PUT("/:id", stationHandler.UpdateRain)
		rain.DELETE("/:id", stationHandler.DeleteRain)
	}

	// Lake Stations
	lake := station.Group("/lake")
	{
		lake.GET("", stationHandler.ListLake)
		lake.GET("/:id", stationHandler.GetLakeByID)
		lake.POST("", stationHandler.CreateLake)
		lake.PUT("/:id", stationHandler.UpdateLake)
		lake.DELETE("/:id", stationHandler.DeleteLake)
	}

	// River Stations
	river := station.Group("/river")
	{
		river.GET("", stationHandler.ListRiver)
		river.GET("/:id", stationHandler.GetRiverByID)
		river.POST("", stationHandler.CreateRiver)
		river.PUT("/:id", stationHandler.UpdateRiver)
		river.DELETE("/:id", stationHandler.DeleteRiver)
	}

	// Pumping Stations
	pumping := station.Group("/pumping")
	{
		pumping.GET("", pumpingHandler.List)
		pumping.GET("/:id", pumpingHandler.Get)
		pumping.POST("", pumpingHandler.Create)
		pumping.PUT("/:id", pumpingHandler.Update)
		pumping.DELETE("/:id", pumpingHandler.Delete)
		pumping.POST("/report", pumpingHandler.CreateHistory)
		pumping.GET("/:id/history", pumpingHandler.ListHistory)
	}

	// Wastewater Treatment Stations
	wastewater := station.Group("/wastewater")
	{
		wastewater.GET("", wastewaterHandler.List)
		wastewater.GET("/:id", wastewaterHandler.Get)
		wastewater.POST("", wastewaterHandler.Create)
		wastewater.PUT("/:id", wastewaterHandler.Update)
		wastewater.DELETE("/:id", wastewaterHandler.Delete)
		wastewater.POST("/:id/report", wastewaterHandler.SubmitReport)
		wastewater.GET("/:id/history", wastewaterHandler.ListHistory)
	}

	// Sluice Gate Stations
	sluice := station.Group("/sluice-gate")
	{
		sluice.GET("", sluiceHandler.List)
		sluice.GET("/:id", sluiceHandler.Get)
		sluice.POST("", sluiceHandler.Create)
		sluice.PUT("/:id", sluiceHandler.Update)
		sluice.DELETE("/:id", sluiceHandler.Delete)
		sluice.POST("/:id/report", sluiceHandler.Report)
		sluice.GET("/:id/history", sluiceHandler.ListHistory)
	}
}

File: ai_tnhn/ai-frontend-tnhn/src/api/station.js
import axiosClient from './axiosClient';

const stationApi = {
    // Rain Stations
    rain: {
        getAll: (params) => axiosClient.get('/admin/stations/rain', { params }),
        getById: (id) => axiosClient.get(`/admin/stations/rain/${id}`),
        create: (data) => axiosClient.post('/admin/stations/rain', data),
        update: (id, data) => axiosClient.put(`/admin/stations/rain/${id}`, data),
        delete: (id) => axiosClient.delete(`/admin/stations/rain/${id}`),
        getHistory: (id, params) => axiosClient.get(`/admin/water/rain/${id}/history`, { params })
    },
    // Lake Stations
    lake: {
        getAll: (params) => axiosClient.get('/admin/stations/lake', { params }),
        getById: (id) => axiosClient.get(`/admin/stations/lake/${id}`),
        create: (data) => axiosClient.post('/admin/stations/lake', data),
        update: (id, data) => axiosClient.put(`/admin/stations/lake/${id}`, data),
        delete: (id) => axiosClient.delete(`/admin/stations/lake/${id}`),
        getHistory: (id, params) => axiosClient.get(`/admin/water/lake/${id}/history`, { params })
    },
    // River Stations
    river: {
        getAll: (params) => axiosClient.get('/admin/stations/river', { params }),
        getById: (id) => axiosClient.get(`/admin/stations/river/${id}`),
        create: (data) => axiosClient.post('/admin/stations/river', data),
        update: (id, data) => axiosClient.put(`/admin/stations/river/${id}`, data),
        delete: (id) => axiosClient.delete(`/admin/stations/river/${id}`),
        getHistory: (id, params) => axiosClient.get(`/admin/water/river/${id}/history`, { params })
    },
    // Inundation Points
    inundation: {
        getAll: (params) => axiosClient.get('/inundation/points-list', { params }),
        create: (data) => axiosClient.post('/inundation/points', data),
        update: (id, data) => axiosClient.put(`/inundation/points/${id}`, data),
        delete: (id) => axiosClient.delete(`/inundation/points/${id}`)
    }
};

export default stationApi;
